src/screens/ProposalDetail.js

```diff
@@ -39,9 +46,21 @@ function ProposalDetail({
 
   const theme = useTheme()
   const panelState = usePanelState()
-  const { vaultBalance } = useAppState()
+  const { vaultBalance, permissions } = useAppState()
+
   const { account: connectedAccount } = useWallet()
 
+  const hasCancelRole = useMemo(() => {
+    if (!connectedAccount) {
+      return false
+    }
+    return permissions.find(
+      ({ roleHash, granteeAddress }) =>
+        roleHash === cancelRoleHash &&
+        addressesEqual(granteeAddress, connectedAccount)
+    )
+  }, [connectedAccount, permissions])
+
```

Comment:
Ah, let's rename the variable to `canCancelProposal` or something around that because the submitter of the proposal can cancel it too without the need of being granted the cancel role, so we should add that logic as well.

Reply:
ah nice